lib/common/s3.sh

```diff
@@ -68,12 +68,32 @@ s3_put_no_index_keep_file() {
     local src=$1; shift
     local object_name=$1; shift
     local dst=$S3_BUCKET/$object_name
+    local mime=''
 
     test -f $src || file_error "Not a regular file"
 
     log_info "Moving '$src' -> '$dst'"
 
-    $S3CMD --no-preserve sync $src $dst || file_error "Could not push to S3 '$src' -> '$dst'"
+    if [[ $file =~ \.gz$ ]]; then
+        mime='application/gzip'
+    elif [[ $file =~ \.nc$ ]]; then
+        mime='application/octet-stream'
+    elif [[ $file =~ \.csv$ ]]; then
+        mime='text/csv'
+    elif [[ $file =~ \.pdf$ ]]; then
+        mime='application/pdf'
+    elif [[ $file =~ \.png$ ]]; then
+        mime='image/png'
+    elif [[ $file =~ \.jpg$ ]]; then
+        mime='image/jpeg'
+    fi
+
+    if [ -z $mime ]; then
+      # use guess mime
+      $S3CMD --no-preserve sync $src $dst || file_error "Could not push to S3 '$src' -> '$dst'"
+    else
+      $S3CMD -m $mime --no-preserve sync $src $dst || file_error "Could not push to S3 '$src' -> '$dst'"
```

Comment:
@julian1 Excellent fix but could we have a complete set of mime types defined? This will be important for the public S3 bucket
.gif .m .jpeg .mov .doc .xls .zip etc etc